java/clients/src/main/java/sleeper/clients/deploy/properties/GeneratePropertiesDocumentation.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.clients.deploy.properties;
+
+import sleeper.clients.util.table.TableWriter;
+import sleeper.core.properties.PropertyGroup;
+import sleeper.core.properties.SleeperProperties;
+import sleeper.core.properties.instance.InstanceProperties;
+import sleeper.core.properties.instance.InstancePropertyGroup;
+import sleeper.core.properties.table.TableProperties;
+import sleeper.core.properties.table.TablePropertyGroup;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.function.Consumer;
+
+public class GeneratePropertiesDocumentation {
+    private static SleeperProperties presentProperties;
+    private static PropertyGroup presentGroup;
```

Comment:
I think it was easier to read when these were passed around as parameters. If we avoid static mutable variables it might be easier to see how these are used?